rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -289,6 +290,19 @@ void SqliteStorage::write(
   commit_transaction();
 }
 
+void SqliteStorage::set_read_order(rosbag2_storage::ReadOrder read_order)
+{
+  read_order_ = read_order;
+  read_statement_ = nullptr;
+
+  if (read_order_.sort_by == rosbag2_storage::ReadOrder::PublishedTimestamp) {
+    throw std::runtime_error("Not Implemented - PublishedTimestamp read order.");
+  }
+  if (read_order_.reverse && seek_row_id_ == 0) {
+    seek_row_id_ = get_last_rowid();
+  }
+}
+
 bool SqliteStorage::has_next()
 {
   if (!read_statement_) {
```

Comment:
@emersonknapp it seems one more bug in `has_next()` method. 
`return current_message_row_ != message_result_.end();` valid only when `reverse==false`

Reply:
Unless I am misunderstanding, I think you've confused the situation.

`message_result_` is a query result iterator, in query order. The query itself is in either `ASC` or `DESC` order. You always increment the iterator to get the next result from the query. When it reaches the `.end()` there are no more results from that query. It doesn't matter what order was specified internal to the query, the iterator always moves `++` towards `.end()`

Reply:
Sorry for the confusion. You are right.